app/repositories/AdminDataRepository.scala

```
@@ -17,101 +17,58 @@
-  } andThen {
-    case _ => CollectionDiagnostics.logCollectionInfo(collection)
-  }
-
-  private def createIndex(indexes: Seq[Index]): Future[Seq[Boolean]] = {
-    Future.sequence(
-      indexes.map { index =>
-        collection.indexesManager.ensure(index) map { result =>
-          logger.debug(s"Index $index was created successfully and result is: $result")
-          result
-        } recover {
-          case e: Exception => logger.error(s"Failed to create index $index", e)
-            false
-        }
-      }
-    )
-  }
 
   def getFeatureToggles: Future[Seq[FeatureToggle]] = {
-
-    implicit val r: Reads[Option[FeatureToggle]] = Reads.optionNoError(FeatureToggle.reads)
-
-    collection
-      .find(
-        selector = BSONDocument("_id" -> featureToggleDocumentId),
-        projection = Option.empty[JsObject]
-      )
-      .one[JsObject].map(_.map(
-      js =>
-        (js \ "toggles")
-          .as[Seq[Option[FeatureToggle]]]
-          .collect {
-            case Some(toggle) => toggle
-          }
-    )).map(_.getOrElse(Seq.empty[FeatureToggle]))
+    collection.find[FeatureToggles](
+      Filters.eq(id, featureToggles)
+    ).headOption().map(_.map(ft =>
+      ft.toggles
+    ).getOrElse(Seq.empty[FeatureToggle])
+    )
   }
 
   def setFeatureToggles(toggles: Seq[FeatureToggle]): Future[Boolean] = {
 
-    val selector = Json.obj(
-      "_id" -> featureToggleDocumentId
-    )
-
-    val modifier = Json.obj(
-      "_id" -> featureToggleDocumentId,
-      "toggles" -> Json.toJson(toggles)
-    )
-
-    collection.update(ordered = false)
-      .one(selector, modifier, upsert = true).map {
-      lastError: WriteResult =>
-        lastError.ok
-    }
+    val upsertOptions = new FindOneAndUpdateOptions().upsert(true)
+    collection.findOneAndUpdate(
+      filter = Filters.eq(id, featureToggles),
+      update = set(featureToggles, Codecs.toBson(toggles)), upsertOptions)
+      .toFuture().map(_ => true)
```

Will this only ever return `true` in all successful futures? e.g. if something goes wrong, will the future fail with an exception rather than returning `false`? If so, would it make more sense for the return type to be `unit`?

---

I think it will always return either true or throw an exception. So yes, I agree a Unit return type would make more sense.